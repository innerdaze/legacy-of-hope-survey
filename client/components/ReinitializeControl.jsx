import React from 'react'
import PropTypes from 'prop-types'
import Menu from 'grommet/components/Menu'
import Anchor from 'grommet/components/Anchor'

const ReinitializeControl = ({ handleReinitializePress = Function.prototype }) => (
  <Menu primary>
    <Anchor
      label='Reconfigure'
      onClick={handleReinitializePress}
      />
  </Menu>
)

ReinitializeControl.propTypes = {
  handleReinitializePress: PropTypes.func.isRequired
}

export default ReinitializeControl
