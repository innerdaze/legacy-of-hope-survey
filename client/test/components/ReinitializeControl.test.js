import React from 'react'
import ReinitializeControl from '../../components/ReinitializeControl'

let wrapper, handleReinitializePress

describe('Testing on ReinitializeControl...', () => {
  beforeEach(() => {
    handleReinitializePress = jest.fn()

    wrapper = shallow(
      <ReinitializeControl handleReinitializePress={handleReinitializePress} />
    )
  })

  describe('ReinitializeControl Layout', () => {
    test('Expect to have a Anchor comp within Menu comp', () => {
      expect(wrapper.find('Menu')).toHaveLength(1)
      expect(wrapper.find('Menu').find('Anchor')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect to call prop function handleReinitializePress if Anchor click', () => {
      wrapper.find('Anchor').simulate('click')

      expect(handleReinitializePress).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
