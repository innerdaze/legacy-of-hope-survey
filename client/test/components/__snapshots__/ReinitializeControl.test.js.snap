// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on ReinitializeControl... ReinitializeControl Layout Expect to have a Anchor comp within Menu comp 1`] = `
<Menu
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  pad="none"
  primary={true}
>
  <Anchor
    animateIcon={true}
    label="Reconfigure"
    method="push"
    onClick={[MockFunction]}
    tag="a"
  />
</Menu>
`;

exports[`Testing on ReinitializeControl... Test on prop Expect to call prop function handleReinitializePress if Anchor click 1`] = `
<Menu
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  pad="none"
  primary={true}
>
  <Anchor
    animateIcon={true}
    label="Reconfigure"
    method="push"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    tag="a"
  />
</Menu>
`;
